controllers/mlmd.go

```diff
@@ -57,7 +60,24 @@ func (r *DSPAReconciler) ReconcileMLMD(dsp *dspav1alpha1.DataSciencePipelinesApp
 			return err
 		}
 	} else {
-		err := r.ApplyDir(dsp, params, mlmdTemplatesDir)
+		err := r.ApplyDir(dsp, params, mlmdTemplatesDir+"/"+mlmdGrpcService)
+		if err != nil {
+			return err
+		}
+
+		if params.PodToPodTLS {
+			var certificatesExist bool
+			certificatesExist, err = params.LoadMlmdCertificates(ctx, r.Client, r.Log)
+			if err != nil {
+				return err
+			}
+
+			if !certificatesExist {
+				return errors.New("secret containing the certificate for MLMD gRPC Server was not created yet")
+			}
+		}
+
+		err = r.ApplyDir(dsp, params, mlmdTemplatesDir)
```

Comment:
(this is really an improvement needed for `ApplyDir`, but until we fix it we should work around it)

The nomeclature of this method doesn't make it obvious that the functionality does _not_ recursively apply all subdirectories.  given that, we should move the files not in the `grpc-service` sudirectory into a seperate one to avoid confusion.  `default` seems to be the de-facto standard here (see other components such as apiserver, for example)

Reply:
(related: created https://issues.redhat.com/browse/RHOAIENG-13196 to address above point)

Reply:
I think he's actually taking advantage of the fact that ApplyDir doesn't recurse. I'm pretty sure he wants the services to go out first on their own. You might then have feelings about using a subdir here if you want to fix up ApplyDir :)